probe-rs/src/debug/variable.rs

```diff
@@ -1,10 +1,319 @@
 use super::*;
+use std::convert::TryInto;
 
-#[derive(Debug, Default)]
+#[derive(Debug, Clone, PartialEq)]
+pub enum VariableKind {
```

Comment:
@Tiwalun ... does this help?
```
///VariableKind is a tag used to differentiate the nature of a variable. The DAP protocol requires a differentiation between 'Named' and 'Indexed'. We are adding 'Referenced', because those require unique handling when decoding the value during runtime
#[derive(Debug, Clone, PartialEq)]
pub enum VariableKind {
    ///An Indexed variable (bound to an ordinal position), such as the sequenced members of an Array or Vector
    Indexed,
    ///A variable that is identified by it's name, and is not bound to a specific ordinal position.
    Named,
    ///A variable that is the target of a pointer variable
    Referenced,
    ///This should never be the final value for a Variable
    Undefined,
}
```

Reply:
Looks good, thanks!